### ISSUE TITLE
iOS push notifications are not being sent to device

### ISSUE THREAD

**Issue by dan-ryan** [fastlane/fastlane#14799]
When setting up Fastlane Match, I broke push notifications (production and staging) on my iOS application (the [nuke][1] command probably did it). Production was logging the errors correctly. 
I generated new certificates with the Fastlane [pem][2] command for both production "com.app.prod" and staging "com.app.qa". It generates new "production" Apple Push Services certificates on the Apple dev site and creates the needed pem files. 

[![enter image description here][3]][3]

I converted the prod pem file to the correct p12 file with the `openssl pkcs12 -export -in "production_com.app.prod.pem" -out "com.app.prod.p12"` command and `openssl pkcs12 -export -in "production_com.app.qa.pem" -out "com.app.qa.p12"` for staging. Uploaded both to AWS SNS service (no sandbox) with no issues. 

[![enter image description here][4]][4]

This fixed the production server. It worked with our old application (before the certificate changes) and the new version. 

But our Testflight staging application was still broken. It did not work with the old or the new version of the iOS application. No errors are coming through. The application registers correctly with the Apple Push Notification service (APNs) and sends the token to our server. Manually pushing a message with an AWS endpoint does not work either but it says it was successfully sent. 

[![enter image description here][5]][5]

Tried regenerating the certificates (made sure I'm uploading the correct ones). I'm using the same iOS device for all my testing. Our Android notifications are still working fine on staging. 

Tried http://pushtry.com, says it was successful but never got any message. 
Also tried the command `openssl s_client -connect gateway.push.apple.com:2195 -cert "production_com.app.qa.pem" -key "production_com.app.qa.pem"`, with no errors returned. 

I'm lost where the issue could be or how I could debug this further. So any suggestions will be helpful.  

  [1]: https://docs.fastlane.tools/actions/match/#nuke
  [2]: https://docs.fastlane.tools/actions/pem/
  [3]: https://i.stack.imgur.com/JIoFP.png
  [4]: https://i.stack.imgur.com/kqEUs.png
  [5]: https://i.stack.imgur.com/QFm0Y.png

**Comment by max-ott** [comment]
`pem` should already provide you with the `p12` files, just to make sure that the issue is not related to the conversion/openssl. Did you try to generate a new certificate? (Maybe revoke the other before, if possible)

As far as I know https://onesignal.com/provisionator uses `pem` (or similar) under the hood. Without any error message it's super hard to debug. :(

**Comment by dan-ryan** [comment]
"pem should already provide you with the p12 files"
It does, but they don't have all the information that AWS needs, so they need to be regenerated.

"just to make sure that the issue is not related to the conversion/openssl"
The instructions I followed is what seems to be common. 

"Did you try to generate a new certificate?"
Yes.

"Maybe revoke the other before, if possible"
Yes, I did use the nuke command again.

"Without any error message it's super hard to debug. :("
Exactly, I'm totally lost where to go from here.
